flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/PulsarSourceEnumerator.java

```diff
@@ -175,11 +175,11 @@ public void close() {
     private void seekStartPosition(Set<TopicPartition> partitions) {
         ConsumerBuilder<byte[]> consumerBuilder =
                 createConsumerBuilder(pulsarClient, Schema.BYTES, configuration);
-        Set<String> seekedPartitions = new HashSet<>();
+        Set<String> seekedTopics = new HashSet<>();
```

Comment:
Since this is a local variable, you still end up seeking the topics on each discovery.
It might be better to switch the call of this function to `assign...` as this is only called on the newly discovered partitions.